hazelcast/src/main/java/com/hazelcast/cardinality/CardinalityEstimator.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cardinality;
+
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.spi.annotation.Beta;
+
+/**
+ * CardinalityEstimator is a redundant and highly available distributed data-structure used
+ * for probabilistic cardinality estimation purposes, on unique items, in significantly sized data cultures.
+ *
+ * CardinalityEstimator is internally based on a HyperLogLog++ data-structure,
+ * and uses P^2 byte registers for storage and computation. (Default P = 14)
+ */
+@Beta
+public interface CardinalityEstimator extends DistributedObject {
+
+    /**
+     * Consider the 64bit hash value, produced by the serialized form of the given object,
+     * in the cardinality estimation. The implementation is free to choose whether this hash
+     * will be used towards the estimation, therefore, this operation may or may not affect
```

Comment:
I am not sure if this detail is so important to have it in the first paragraph. 

This is the most important method of the whole interface. It should have a clear API which states: "Use this to add a new object into your estimation". Something along these lines. It should be the first sentence - for super-quick discoverability. Everything else is just a detail/clarification.